### PR TITLE
docs(prefer-screen-queries): update allowed properties and methods

### PR DESCRIPTION
**Pull request by gndelia**
Requires rebasing after #157  is merged. The changes are in this commit f17275e

edit: The properties + methods I included come from the typings here https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/testing-library__react/index.d.ts#L19

### CODE REVIEW

**Comment by Belco90** [comment]
Nice one! #157 is merged so you can rebase it now.

File: tests/lib/rules/prefer-screen-queries.test.ts [hunk 1]
```diff
@@ -33,6 +33,36 @@ ruleTester.run(RULE_NAME, rule, {
         myWithinVariable.${queryMethod}('baz')
       `,
     })),
+    {
+      code: `
+        const screen = render(baz);
+        screen.container.querySelector('foo');
```
Comment (Belco90) [inline tests/lib/rules/prefer-screen-queries.test.ts:39]: This is a good edge case 💯

File: docs/rules/prefer-screen-queries.md [hunk 2]
```diff
@@ -2,8 +2,8 @@
 
 ## Rule Details
 
-DOM Testing Library (and other Testing Library frameworks built on top of it) exports a `screen` object which has every query (and a `debug` method). This works better with autocomplete and makes each test a little simpler to write and maintain.
-This rule aims to force writing tests using queries directly from `screen` object rather than destructuring them from `render` result.
+DOM Testing Library (and other Testing Library frameworks built on top of it) exports a `screen` object which has every query (and a `debug` method). This works better with autocomplete and makes each test a little simpler to write and maintain.  
+This rule aims to force writing tests using queries directly from `screen` object rather than destructuring them from `render` result. Given the screen component does not expose utility methods such as `rerender()` or the `container` property, it is correct to use the `render` response in those scenarios-
```
Comment (Belco90) [inline docs/rules/prefer-screen-queries.md:6]: There is a weird dash at the end of the sentence. Could you fix it please?

Comment (gndelia) [reply docs/rules/prefer-screen-queries.md]: uops, typo

Comment (gndelia) [reply docs/rules/prefer-screen-queries.md]: fixed
